packages/compass-saved-aggregations-queries/src/components/open-item-modal.tsx

```diff
@@ -0,0 +1,198 @@
+import React from 'react';
+import {
+  ConfirmationModal,
+  Option,
+  Select,
+  css,
+  spacing,
+} from '@mongodb-js/compass-components';
+import { connect, MapDispatchToProps, MapStateToProps } from 'react-redux';
+import { RootState } from '../stores';
+import {
+  closeModal,
+  openSelectedItem,
+  selectCollection,
+  selectDatabase,
+} from '../stores/open-item';
+
+type AsyncItemsSelectProps = {
+  name: string;
+  label: string;
+  items: string[];
+  selectedItem: string | null;
+  isLoading: boolean;
+  onChange(item: string): void;
+};
+
+const select = css({
+  // Working around leafygreen Select issues
+  // See (TODO: link to JIRA ticket)
+  button: {
+    zIndex: 999,
+  },
+});
+
+const AsyncItemsSelect: React.FunctionComponent<AsyncItemsSelectProps> = ({
+  name,
+  label,
+  items,
+  selectedItem,
+  isLoading,
+  onChange,
+}) => {
+  return (
+    <Select
+      name={name}
+      label={label}
+      value={selectedItem ?? ''}
+      onChange={onChange}
+      disabled={isLoading}
+      usePortal={false}
+      className={select}
+    >
+      {items.map((item) => (
+        <Option key={item} value={item}>
+          {item}
+        </Option>
+      ))}
+    </Select>
+  );
+};
+
+const mapDatabaseState: MapStateToProps<
+  Pick<AsyncItemsSelectProps, 'items' | 'selectedItem' | 'isLoading'>,
+  Pick<AsyncItemsSelectProps, 'name' | 'label'>,
+  RootState
+> = ({ openItem: { databases, databasesStatus, selectedDatabase } }) => {
+  return {
+    items: databases,
+    selectedItem: selectedDatabase,
+    isLoading: ['initial', 'fetching'].includes(databasesStatus),
+  };
+};
+
+const DatabaseSelect = connect(mapDatabaseState, { onChange: selectDatabase })(
+  AsyncItemsSelect
+);
+
+const mapCollectionState: MapStateToProps<
+  Pick<AsyncItemsSelectProps, 'items' | 'selectedItem' | 'isLoading'>,
+  Pick<AsyncItemsSelectProps, 'name' | 'label'>,
+  RootState
+> = ({ openItem: { collections, collectionsStatus, selectedCollection } }) => {
+  return {
+    items: collections,
+    selectedItem: selectedCollection,
+    isLoading: ['initial', 'fetching'].includes(collectionsStatus),
+  };
+};
+
+const CollectionSelect = connect(mapCollectionState, {
+  onChange: selectCollection,
+})(AsyncItemsSelect);
+
+type OpenItemModalProps = {
+  namespace: string;
+  itemType: string;
+  itemName: string;
+  isModalOpen: boolean;
+  isSubmitDisabled: boolean;
+  onSubmit(): void;
+  onClose(): void;
+};
+
+const modalContent = css({
+  display: 'grid',
+  gridTemplateAreas: `
+    'description description'
+    'database collection'
+  `,
```

Comment:
Worth noting that we haven't really discussed this layout yet, but this is more or less the only easy way to work around leafygreen issues with select components and zIndex: having them side by side means that components are not running into the chance of overlapping with each other and can use the same zIndex to be on top of the buttons in the modal footer